hstream/src/HStream/Server/Handler/Common.hs

```diff
@@ -128,3 +131,15 @@ handleCreateAsSelect ServerContext{..} taskBuilder commandQueryStmtText extra =
 
 mark :: (Exception e, Exception f) => (e -> f) -> IO a -> IO a
 mark mke = handle (throwIO . mke)
+
+handleTerminateConnector :: ServerContext -> CB.CBytes
+  -> IO ()
+handleTerminateConnector ServerContext{..} cid = do
+  hmapC <- readMVar runningConnectors
+  case HM.lookup cid hmapC of
+    Just tid -> killThread tid
+    -- TODO: shall we throwIO here
+    _        -> return ()
+  -- TODO: shall we move this op to Just tid -> killThread tid
```

Comment:
What do these two TODOs mean?

Reply:
I'm thinking about if we can't find cid in map. Do we need to throw error there.
And if we can't find cid in map, we should still update the state in zookeeper? 
The real problem is that shall we distinguish the cancel and the terminate. I will open an issue.

Reply:
And also, the restart logic is not full implemented yet. I write it in this way to pass the tests.

Reply:
For example? Under which circumstance

Reply:
After remove cid from MVar. We didn't update status in ZooKeeper successfully. Somehow we try to terminate it again. 
I think we should update the status in ZooKeeper.
Anyway... Terminate should be a idempotent operation, I guess?